From 532b45eca70213fdfe09b45a1d302bee5d0b2391 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Le=20Goffic?= <clement.legoffic@foss.st.com>
Date: Tue, 6 Feb 2024 09:45:22 +0100
Subject: [PATCH 1138/1695] drivers: stm32_rtc: remove includes and reorder
 macros definition
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Reorder the register macros with the order we found in the
documentation of RTC register.
Add documentation header for the macros.
Remove unused header files inclusion.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git ac406cb6e132bf5c66510c2654276865e6d5e2e4 drivers: stm32_rtc: remove includes and reorder macros definition]
Change-Id: Ie4073910247a0eb6261d1cd2553bd1f7f7fa29cc
Signed-off-by: Clément Le Goffic <clement.legoffic@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/402351
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/stm32_rtc.c | 70 +++++++++++++++++++++-------------------
 1 file changed, 36 insertions(+), 34 deletions(-)

diff --git a/core/drivers/stm32_rtc.c b/core/drivers/stm32_rtc.c
index 852931292..1067c4ae2 100644
--- a/core/drivers/stm32_rtc.c
+++ b/core/drivers/stm32_rtc.c
@@ -8,21 +8,19 @@
 #else
 #include <arm32.h>
 #endif /* CFG_ARM64_core */
-#include <config.h>
 #include <drivers/clk.h>
 #include <drivers/clk_dt.h>
 #include <drivers/stm32_rtc.h>
 #include <drivers/stm32_rif.h>
-#include <initcall.h>
 #include <io.h>
-#include <kernel/boot.h>
-#include <kernel/delay.h>
 #include <kernel/dt.h>
 #include <kernel/panic.h>
 #include <libfdt.h>
 #include <mm/core_memprot.h>
-#include <stm32_util.h>
 
+/*
+ * Registers
+ */
 #define RTC_TR				U(0x00)
 #define RTC_DR				U(0x04)
 #define RTC_SSR				U(0x08)
@@ -87,11 +85,24 @@
 #define RTC_CR_BYPSHAD_SHIFT		U(5)
 #define RTC_CR_TAMPTS			BIT(25)
 
+#define RTC_PRIVCFGR_VALUES		GENMASK_32(3, 0)
+#define RTC_PRIVCFGR_VALUES_TO_SHIFT	GENMASK_32(5, 4)
+#define RTC_PRIVCFGR_SHIFT		U(9)
+#define RTC_PRIVCFGR_MASK		(GENMASK_32(14, 13) | GENMASK_32(3, 0))
+#define RTC_PRIVCFGR_FULL_PRIV		BIT(15)
+
 #define RTC_SMCR_TS_DPROT		BIT(3)
-#define RTC_SR_TSF			BIT(3)
-#define RTC_SCR_CTSF			BIT(3)
-#define RTC_SR_TSOVF			BIT(4)
-#define RTC_SCR_CTSOVF			BIT(4)
+
+#define RTC_SECCFGR_VALUES		GENMASK_32(3, 0)
+#define RTC_SECCFGR_TS_SEC		BIT(3)
+#define RTC_SECCFGR_VALUES_TO_SHIFT	GENMASK_32(5, 4)
+#define RTC_SECCFGR_SHIFT		U(9)
+#define RTC_SECCFGR_MASK		(GENMASK_32(14, 13) | GENMASK_32(3, 0))
+#define RTC_SECCFGR_FULL_SEC		BIT(15)
+
+#define RTC_WPR_KEY1			U(0xCA)
+#define RTC_WPR_KEY2			U(0x53)
+#define RTC_WPR_KEY_LOCK		U(0xFF)
 
 #define RTC_TSDR_MU_MASK		GENMASK_32(11, 8)
 #define RTC_TSDR_MU_SHIFT		U(8)
@@ -100,40 +111,20 @@
 #define RTC_TSDR_DU_MASK		GENMASK_32(3, 0)
 #define RTC_TSDR_DU_SHIFT		U(0)
 
-#define RTC_WPR_KEY1			U(0xCA)
-#define RTC_WPR_KEY2			U(0x53)
-#define RTC_WPR_KEY_LOCK		U(0xFF)
+#define RTC_SR_TSF			BIT(3)
+#define RTC_SR_TSOVF			BIT(4)
+
+#define RTC_SCR_CTSF			BIT(3)
+#define RTC_SCR_CTSOVF			BIT(4)
 
 #define RTC_CIDCFGR_SCID_MASK		GENMASK_32(6, 4)
 #define RTC_CIDCFGR_SCID_MASK_SHIFT	U(4)
 #define RTC_CIDCFGR_CONF_MASK		(_CIDCFGR_CFEN |	 \
 					 RTC_CIDCFGR_SCID_MASK)
 
-#define RTC_PRIVCFGR_FULL_PRIV		BIT(15)
-#define RTC_PRIVCFGR_VALUES		GENMASK_32(3, 0)
-#define RTC_PRIVCFGR_VALUES_TO_SHIFT	GENMASK_32(5, 4)
-#define RTC_PRIVCFGR_SHIFT		U(9)
-#define RTC_PRIVCFGR_MASK		(GENMASK_32(14, 13) | GENMASK_32(3, 0))
-
-#define RTC_SECCFGR_FULL_SEC		BIT(15)
-#define RTC_SECCFGR_VALUES		GENMASK_32(3, 0)
-#define RTC_SECCFGR_VALUES_TO_SHIFT	GENMASK_32(5, 4)
-#define RTC_SECCFGR_SHIFT		U(9)
-#define RTC_SECCFGR_TS_SEC		BIT(3)
-#define RTC_SECCFGR_MASK		(GENMASK_32(14, 13) | GENMASK_32(3, 0))
-
-#define RTC_RES_TIMESTAMP		U(3)
-
-#define RTC_FLAGS_READ_TWICE		BIT(0)
-#define RTC_FLAGS_SECURE		BIT(1)
-
-#define TIMEOUT_US_RTC_SHADOW		U(10000)
-#define MS_PER_SEC			U(1000)
-
 /*
  * RIF miscellaneous
  */
-
 #define RTC_NB_RIF_RESOURCES		U(6)
 
 #define RTC_RIF_FULL_PRIVILEGED		U(0x3F)
@@ -143,6 +134,17 @@
 
 #define RTC_CID_1			U(1)
 
+/*
+ * Driver miscellaneous
+ */
+#define RTC_RES_TIMESTAMP		U(3)
+
+#define RTC_FLAGS_READ_TWICE		BIT(0)
+#define RTC_FLAGS_SECURE		BIT(1)
+
+#define TIMEOUT_US_RTC_SHADOW		U(10000)
+#define MS_PER_SEC			U(1000)
+
 struct rtc_compat {
 	bool has_seccfgr;
 	bool has_rif_support;
-- 
2.47.2

